### DISCUSSION TITLE
Required vs Present rule by default

### DISCUSSION THREAD

**Discussion by iamrgroot** [spatie/laravel-data#108] [Q&A] [answered]
With the current implementation a DataCollection will get the 'required' rule added by default. However, this rule disallows for empty values: https://laravel.com/docs/8.x/validation#rule-required . So sending an empty array to a data class property of type DataCollection will not validate. 

Is this by design? Of would the 'present' rule be a better option by default https://laravel.com/docs/8.x/validation#rule-present ?

I also can't seem to use a Rule Attribute when using Data/DataCollection, as the RuleInferrerers only get run when the property is not Data and not DataCollection... Is this correct? Could this be changed? Using the rules() function does work but I would prefer an Attribute

**Comment by rubenvanassche** [comment, marked as answer]
@IAmRGroot fixed this in v1.4.4. 

What validation attributes would you add to a data property?

**Reply by iamrgroot** [reply]
This was mostly regarding the DataCollection not allowing empty arrays, so as a solution I tried to add the Present rule as an attribute, but it stayed required as the Required rule didn't check for a Present rule before inferring it.

**Reply by rubenvanassche** [reply]
Yeah, by default we will nog use `present` for collections. You still cannot add extra validateable attributes to data objects or collections because I cannot find any use case for it 😄
